### PR TITLE
change group migrate to deal with long-running imports

### PR DESCRIPTION
**Pull request by malaffoon**

### CODE REVIEW

File: migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml [hunk 1]
```diff
@@ -648,34 +652,13 @@ sql:
                 wsg.deleted,
                 wsg.updated
               FROM student_group wsg
-              WHERE wsg.created > :first_at AND wsg.created <= :last_at
+                JOIN import i ON i.id = wsg.update_import_id
+              WHERE i.updated > :first_at AND i.updated <= :last_at
```
Comment (malaffoon) [inline migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml]: This is the key part of the change: use the import records final timestamp to pull in all groups from that import.

File: migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml [hunk 2]
```diff
@@ -648,34 +652,13 @@ sql:
...
 
             stagingInsert: >-
               INSERT INTO staging_student_group (id, name, school_id, school_year, subject_id, active, creator, created, update_import_id, deleted, updated, migrate_id) VALUES
                 (:id, :name, :school_id, :school_year, :subject_id, :active, :creator, :created, :update_import_id, :deleted, :updated, :migrate_id)
 
-        student_group_by_update_import_id:
```
Comment (malaffoon) [inline migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml]: @agorina i'd like to discuss this. I don't see any purpose in separating new groups from updated groups at this point.

File: migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml [hunk 3]
```diff
@@ -686,15 +669,16 @@ sql:
               FROM student_group_membership wsgm
                 JOIN student_group wsg ON wsg.id = wsgm.student_group_id
                 JOIN student ws ON ws.id = wsgm.student_id
-              WHERE ( (wsg.created > :first_at AND wsg.created <= :last_at) OR (wsg.updated > :first_at AND wsg.updated <= :last_at) )
+                JOIN import i ON i.id = wsg.update_import_id
+              WHERE i.updated > :first_at AND i.updated <= :last_at
```
Comment (malaffoon) [inline migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml]: Use the same criteria for group membership (users and students).

File: migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml (hunk 2 above)
Comment (agorina) [reply migrate-reporting/src/main/resources/application.warehouse.to.staging.sql.yml]: Makes sense to me.

**Pull request merged**